src/app/pages/examiner-stats/examiner-stats.page.ts

```diff
@@ -0,0 +1,391 @@
+        manoeuvres$: this.filterByParameters(getManoeuvresUsed),
+        safetyAndBalanceQuestions$: this.filterByParameters(getSafetyAndBalanceQuestions),
+        independentDriving$: this.filterByParameters(getIndependentDrivingStats),
+        showMeQuestions$: this.filterByParameters(getShowMeQuestions),
+        tellMeQuestions$: this.filterByParameters(getTellMeQuestions),
+        testCount$: this.filterByParameters(getStartedTestCount),
+        outcomes$: this.filterByParameters(getOutcome),
+        locationList$: this.filterByParameters(getLocations),
+        categoryList$: this.filterByParameters(getCategories),
+        controlledStops$: this.filterByParameters(getStartedTestCount)
+          .pipe(
+            withLatestFrom(this.filterByParameters(getControlledStopCount)),
+            map(([testCount, controllerStopCount]) => ([
+              {
+                item: 'Controlled stop',
+                count: controllerStopCount,
+                percentage: `${((controllerStopCount / testCount) * 100).toFixed(1)}%`,
+              },
+              {
+                item: 'No controlled stop',
+                count: testCount - controllerStopCount,
+                percentage: `${(((testCount - controllerStopCount) / testCount) * 100).toFixed(1)}%`,
+              },
+            ])),
+          ),        passPercentage$: this.filterByParameters(getStartedTestCount)
+          .pipe(
+            withLatestFrom(
+              this.filterByParameters(getPassedTestCount),
+            ),
+            map(this.calculatePercentage),
+          ),
+      };
+
+      this.setFilterLists();
+    }
+
+    ionViewDidEnter() {
+      this.store$.dispatch(ExaminerStatsViewDidEnter());
+    }
+
+    setFilterLists() {
+      if (!this.locationFilterOptions) {
+        this.locationFilterOptions = [];
+        let locationList = [];
+
+        this.pageState.locationList$.subscribe(value => {
+          locationList = value;
+        })
+          .unsubscribe();
+
+        let tempArray: TestCentre[] = [];
+        locationList.forEach((val) => {
+          tempArray.push(val.item);
+        });
+
+        this.locationFilterOptions = tempArray;
+        let mostUsed = this.setDefault(locationList);
+        this.locationPlaceholder = mostUsed.item.centreName;
+        this.handleLocationFilter(mostUsed.item);
+      }
+      if (!this.categoryFilterOptions) {
+        this.categoryFilterOptions = [];
+        let tempCatList = [];
+
+        this.pageState.categoryList$.subscribe(value => {
+          tempCatList = value;
+        })
+          .unsubscribe();
+
+        let tempArray: TestCategory[] = [];
+        tempCatList.forEach((val) => {
+          tempArray.push(val.item);
+        });
+
+        this.categoryFilterOptions = tempArray;
+        let mostUsed = this.setDefault(tempCatList);
+        this.categoryPlaceholder = mostUsed.item;
+        this.handleCategoryFilter(mostUsed.item);
```

Comment:
I think we could spend some time looking to see if we can optimise/stream line this code. Im not a fan of calling anything `temp`, and there seems to be a lot of similarity between the two paths.